docs/docs/guide/project/dependencies.md

```diff
@@ -244,6 +244,55 @@ pod install
 > [!WARNING]
 > CocoaPods dependencies are not compatible with workflows like `build` or `test` that run `xcodebuild` right after generating the project. They are also incompatible with binary caching and selective testing since the fingerprinting logic doesn't account for the Pods dependencies.
 
+## Static or dynamic
+
+Frameworks and libraries can be linked either statically or dynamically, a choice that has significant implications for aspects like app size and boot time. Despite its importance, this decision is often made without much consideration. Tools like [Swift Package Manager](https://www.swift.org/documentation/package-manager/) and [Mergeable Libraries](https://developer.apple.com/documentation/xcode/configuring-your-project-to-use-mergeable-libraries) automatically make this choice during build-time, which, while convenient, adds complexity to the build process and introduces potential sources of non-determinism. This can cause some Xcode features, such as SwiftUI previews, to become unreliable.
+
+**Our recommendation is to trade some build-time convenience for the reliability and determinism of making this decision yourself.** By doing so, Xcode will work more reliably with your project, and your builds will be more predictable.
+
+The simplest approach is to link everything statically in release builds for faster booting and dynamically in debug builds for faster iteration. You can use [dynamic configurations](http://localhost:5173/guide/project/dynamic-configuration.html#reading-the-environment-variables-from-manifests) to manage this at generation-time. You need to include a helper function in your manifest files that returns the appropriate product type based on the environment variable `TUIST_LINKING`:
```

Comment:
This gets more complicated for any apps that include extensions.  The consequence of release static linking would include all dependencies bytecode into extensions that depend on shared frameworks, bloating app size.  I think we should probably provide a set of scenarios and what we feel are good practices in each vs a one sized "simplest" recommendation.

Reply:
This is a good one. I'll add an example. Can you think of other scenarios?

Reply:
There's also the added complexity of linker flags being needed if the dependencies contain Objective-C code as well.

Reply:
@waltflanagan does this apply to frameworks as well? Given an example where all my custom frameworks if made static, and they all depend on dynamically configured frameworks, would that in effect duplicate and bloat the overall app size?

Reply:
@pepicrft @waltflanagan Not sure if you want to add it as an example, but for example in a large codebase with a large set of targets you probably have some core level frameworks which are linked into a lot of the upper frameworks. To avoid duplication I think dynamic frameworks are preferable over static ones in this scenario. Actually similar to the extension case

Reply:
I would have to investigate with Static Frameworks and their behavior.  It could be that those do solve for the issue. If so we would need to call out the entire framework vs library and dynamic vs static issue.